tests/test_order.py

```diff
@@ -0,0 +1,201 @@
+    def test_read_order(self):
+        """It should Read an order"""
+        order = OrderFactory()
+        order.create()
+
+        # Read it back
+        found_order = Order.find(order.id)
+        self.assertIsNotNone(found_order)
+        self.assertEqual(found_order.id, order.id)
+        self.assertEqual(found_order.customer_name, order.customer_name)
+        self.assertEqual(found_order.status, order.status)
+        self.assertEqual(found_order.created_at, order.created_at)
+        self.assertEqual(found_order.updated_at, order.updated_at)
+        self.assertEqual(len(found_order.items), 0)
+
+    def test_list_all_orders(self):
+        """It should List all Orders in the database"""
+        orders = Order.all()
+        self.assertEqual(orders, [])
+        for order in OrderFactory.create_batch(5):
+            order.create()
+        # Assert that there are 5 orders in the database
+        orders = Order.all()
+        self.assertEqual(len(orders), 5)
+
+    def test_find_by_name(self):
+        """It should Find an Order by name"""
+        order = OrderFactory()
+        order.create()
+
+        # Fetch it back by name
+        same_order = Order.find_by_name(order.customer_name)[0]
+        self.assertEqual(same_order.id, order.id)
+        self.assertEqual(same_order.customer_name, order.customer_name)
+
+    def test_serialize_an_order(self):
+        """It should Serialize an order"""
+        order = OrderFactory()
+        item = ItemFactory()
+        order.items.append(item)
+        serial_order = order.serialize()
+        self.assertEqual(serial_order["id"], order.id)
+        self.assertEqual(serial_order["customer_name"], order.customer_name)
+        self.assertEqual(serial_order["status"], order.status)
+        self.assertEqual(serial_order["created_at"], str(order.created_at))
+        self.assertEqual(serial_order["updated_at"], str(order.updated_at))
+        self.assertEqual(len(serial_order["items"]), 1)
+        items = serial_order["items"]
+        self.assertEqual(items[0]["id"], item.id)
+        self.assertEqual(items[0]["order_id"], item.order_id)
+        self.assertEqual(items[0]["product_name"], item.product_name)
+        self.assertEqual(items[0]["quantity"], item.quantity)
+        self.assertEqual(items[0]["price"], item.price)
+        self.assertEqual(items[0]["created_at"], item.created_at.isoformat())
+        self.assertEqual(items[0]["updated_at"], item.updated_at.isoformat())
+
+    def test_deserialize_an_order(self):
+        """It should Deserialize an order"""
+        order = OrderFactory()
+        print("order from order factory")
+        print(order)
+        order.items.append(ItemFactory())
+        order.create()
+        serial_order = order.serialize()
+        new_order = Order()
+        
+        new_order.deserialize(serial_order)
+        print("deserialized order from order factory")
+        print(new_order)
+
+        self.assertEqual(new_order.id, order.id)
```

Comment:
We shouldn't check "order.id" and "item.id" during deserialization. 
When a user sends a create request via a URL, we need to deserialize the user's request into an object. While, the user do not know the object's ID in advance because it hasn't been created yet XD.

[Professor's example account.py line72](https://github.com/nyu-devops/sample-accounts/blob/master/service/models/account.py)
[Professor's example test_account.py line 216](https://github.com/nyu-devops/sample-accounts/blob/master/tests/test_account.py)

Reply:
oh I see. I was wondering why they were missing in deserialization but thought it could be a typo. Good catch and thanks for telling me the reason. It makes sense. WIll remove it from both places.